### PR TITLE
Merge changes from our midstream that need to also be upstream

### PR DESCRIPTION
**Pull request by gparvin**
See the contained commits for the details.

### CODE REVIEW

**Comment by dhaiducek** [comment]
@gparvin Looks like there are some linting errors here.

File: build/common/config/.golangci.yml [hunk 1]
```diff
@@ -53,6 +67,8 @@ linters-settings:
   govet:
     # report about shadowed variables
     check-shadowing: false
+  gci:
+    local-prefixes: github.com/open-cluster-management/governance-policy-template-sync
```
Comment (dhaiducek) [inline build/common/config/.golangci.yml:71]: Ah, this is the spot. 🙂

File: Makefile [hunk 2]
```diff
@@ -213,18 +243,19 @@ kind-delete-cluster:
 
 install-crds:
 	@echo installing crds
-	kubectl apply -f https://raw.githubusercontent.com/open-cluster-management-io/governance-policy-propagator/main/deploy/crds/policy.open-cluster-management.io_policies.yaml
+	kubectl apply -f https://raw.githubusercontent.com/open-cluster-management-io/governance-policy-propagator/$(BRANCH)/deploy/crds/policy.open-cluster-management.io_policies.yaml
+	kubectl apply -f https://raw.githubusercontent.com/stolostron/config-policy-controller/$(BRANCH)/deploy/crds/policy.open-cluster-management.io_configurationpolicies.yaml
```
Comment (mprahl) [inline Makefile:247]:
```suggestion
	kubectl apply -f https://raw.githubusercontent.com/open-cluster-management/config-policy-controller/$(BRANCH)/deploy/crds/policy.open-cluster-management.io_configurationpolicies.yaml
```

File: build/common/config/.golangci.yml [hunk 3]
```diff
@@ -164,13 +180,32 @@ issues:
...
-    # Exclude some linters from running on test files.
-    - path: _test\.go$|^tests/|^samples/
+    # Allow dot imports in the tests.
+    - path: _test\.go$|^test/
+      linters:
+        - gci
+        - golint
+        - revive
+        - stylecheck
+      source: \. "github\.com/onsi/(gomega|ginkgo)"
```
Comment (mprahl) [inline build/common/config/.golangci.yml:195]: Other repos have it like this. I'm not sure if that was missed in the merge conflicts.
```suggestion
      source: \. "github\.com/onsi/(gomega|ginkgo/v2)"
```

File: build/common/config/.golangci.yml (hunk 1 above)
Comment (dhaiducek) [reply build/common/config/.golangci.yml:71]: Actually, that not might be it. Double checking.

Comment (dhaiducek) [reply build/common/config/.golangci.yml:71]: Yup, I had the path wrong--I've updated the suggestion.

File: Makefile (hunk 2 above)
Comment (dhaiducek) [reply Makefile:247]: This needs the `-io` suffix also

File: build/common/config/.golangci.yml (hunk 1 above)
Comment (gparvin) [reply build/common/config/.golangci.yml]: Thanks, update made

File: Makefile (hunk 2 above)
Comment (gparvin) [reply Makefile]: Yes, got it.  Updated.

File: build/common/config/.golangci.yml (hunk 3 above)
Comment (gparvin) [reply build/common/config/.golangci.yml]: I've updated this.  Thanks!

**Review by dhaiducek — APPROVED** [review]

**Pull request merged**